### ISSUE TITLE
extend predictive algorithms/models for EHPA

### ISSUE THREAD

**Issue by ghost** [gocrane/crane#397]
## Describe the feature
Currently, EHPA has only supported the DSP kind of algorithms, it would be better if we could support much broader types of predictive models.

## Motivation
As feature request [#374](https://github.com/gocrane/crane/issues/374) suggested, EHPA might would support prediction based on external metrics in addition to the current CPU; along with more metrics being integrated in the future, naturally; more predictive models are needed since more time-series patterns are coming, which could not be captured in the best way by only one algorithm family (e.g., DSP);

On the other way, we can compare the performance of different models, to find the best offer no the same time series data;

As the first suggestion, I propose to add the Time2Graph model into EHPA; Time2Graph has been demonstrated on considerable datasets in time-series prediction and shown a significant advantage.
For details: https://arxiv.org/abs/1911.04143

**Comment by xieydd** [comment]
Time series prediction algorithm has long history. Up to now, there have been many research results in this field, including the current use of deep learning algorithms such as transformer for time series prediction. Our proposed dsp mainly considers interpretability in production environment. But we are also actively exploring some of the latest research on time series forecasting. If you are interested, we can discuss it together. If possible, can you send an email to me? :  )
@ClouderPing

**Comment by ClouderPing** [comment]
Yes, time-series prediction does have long history; the question is, what kind of predictive algorithms are suitable or effective for cloud native workload pattern learning? there are considerable interpretable algorithms and DSP is one of them; EHPA should support at least several types of models; but more, it needs a light-weight mechanism for comparing the performance of multiple models when users are targeting on the same workload data; we can keep the part of the predictive algorithms as light as possible but meanwhile support for model selection to some extent. 

Actually, making those algorithms as a plug-in might be the best choice. Crane can provide only the well-defined interfaces and let users implement the specific models in interest themselves;

**Comment by ClouderPing** [comment]
Have sent email to you. THANKS
